enzy_htp/structure/structure_region/residue_caps.py

```diff
@@ -0,0 +1,579 @@
+        'HP13': 0.0,
+        # C-terminal
+        "NP2": 0.0,
+        "HP2": 0.0,
+        "CP2": 0.0,
+        "HP21": 0.0,
+        "HP22": 0.0,
+        "HP23": 0.0,
+    }
+    """Fixed charge for a NHCH3 attached to the N/C-ter of a Residue()."""
+
+
+    CAP_BOND_DISTANCE:Dict[str, float] = {
+        'N': 1.33,
+        'C': 1.33
+    }
+    """Maps the name of the link_atom to appropriate bond distance."""
+
+    def get_nterm_atoms(self) -> List[Atom]:
+        """Create the default n-terminal version of the NHCH3Cap with appropriate names."""
+        return [
+            Atom(name='NP1',  coord=[ 0.000,   0.000,  0.000], element='N'),
+            Atom(name='HP1',  coord=[ -0.5,   -0.860,  0.000], element='H'),
+            Atom(name='CP1',  coord=[ 1.500,   0.000,  0.000], element='C'),
+            Atom(name='HP11', coord=[ 1.864,   0.000, -1.027], element='H'),
+            Atom(name='HP12', coord=[ 1.864,  -0.889,  0.515], element='H'),
+            Atom(name='HP13', coord=[ 1.864,   0.889,  0.515], element='H'),
+        ]
+
+
+    def get_cterm_atoms(self) -> List[Atom]:
+        """Create the default c-terminal version of the NHCH3Cap with appropriate names."""
+        return [
+            Atom(name='NP2',  coord=[ 0.000,   0.000,  0.000], element='N'),
+            Atom(name='HP2',  coord=[ -0.5,   -0.860,  0.000], element='H'),
+            Atom(name='CP2',  coord=[ 1.500,   0.000,  0.000], element='C'),
+            Atom(name='HP21', coord=[ 1.864,   0.000, -1.027], element='H'),
+            Atom(name='HP22', coord=[ 1.864,  -0.889,  0.515], element='H'),
+            Atom(name='HP23', coord=[ 1.864,   0.889,  0.515], element='H'),
+        ]
+
+    def net_charge(self) -> int:
+        """Net charge always zero for this neutral group."""
+        return 0
+
+    def anchor(self) -> None:
+        """Straightforward anchoring that is mostly identifcal for both n-terminal and c-terminal capping. Will warn when 
+        user attempts to cap with NHCH3 on n-terminal side as this creates an unusual and unrealistic NH-NH bond."""
+
+        if self.is_nterm_cap():
+            _LOGGER.warning("WARNING: You are trying to add a methylamide cap on the n-terminal side of an amino acid!")
+        
+        p0 = np.array(self.link_atom.coord)
+        p1 = np.array(self.socket_atom.coord)
+        p2 = np.array(self.socket_atom.parent.find_atom_name('CA').coord)
+
+        direction0 = p1 - p0        
+        direction0 /= np.linalg.norm(direction0)
+        
+        direction = p2 - p1
+        direction /= np.linalg.norm(direction)
+        rot_mat = rotation_matrix_from_vectors(np.array([1,0,0]), direction)
+        
+        cap_bond_distance = self.CAP_BOND_DISTANCE[self.link_atom.name]
+
+        for aa in self.atoms:
+            pt = np.array(aa.coord)
+            pt = np.transpose(np.matmul(rot_mat, np.transpose( pt  )))
+            pt += (direction0*cap_bond_distance) + p0
+            aa.coord = pt
+
```

Comment:
We would like to reuse these from math_helper as they are repetative for most anchor

Reply:
See above comments.